scram-tools.file/tools/gcc/gcc-cxxcompiler.xml

```diff
@@ -9,7 +9,7 @@
     </client>
     <flags CPPDEFINES="GNU_GCC _GNU_SOURCE @GCC_CPPDEFINES@"/>
     <flags CXXSHAREDOBJECTFLAGS="-fPIC @GCC_CXXSHAREDOBJECTFLAGS@"/>
-    <flags CXXFLAGS="-O3 -pthread -pipe -Werror=main -Werror=pointer-arith"/>
+    <flags CXXFLAGS="-O3 -g -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -pthread -pipe -Werror=main -Werror=pointer-arith"/>
```

Comment:
@AuroraPerego , we are planning to build a dedicated IB with frame pointers. I will setup an IB for this. 
By the way, why enable `-g` ?

Reply:
@smuzaffar, debug symbols are needed for getting function names and source code lines and paths (if available) for every stack trace obtained during profiling.

Reply:
`-g` can not go in to production builds . In past it also broke other optimizations

Reply:
Ciao @smuzaffar I vaguely recall we used to have a separate build with all debugging symbols?

Reply:
yes, we still have e.g. `CMSSW_15_1_DBG_X` IBs

Reply:
can we add frame pointers there as well?

Reply:
sure we can, let s first wait how enabling frame-pointer behaves in `FP_X` IBs